File: frontend/src/models/Message.ts
import User from "./User";

export default class Message {
    id: string;
    content: string;
    sentTime: Date;
    sender: User;
    receiver: User;
    isRead: boolean;

    constructor(content: string, id: string, sentTime: Date, sender: User, receiver: User, isRead: boolean) {
        this.id = id;
        this.content = content;
        this.sentTime = sentTime;
        this.sender = sender;
        this.receiver = receiver;
        this.isRead = isRead;
    }
};

File: Backend/Backend/Domain/Entity/Message.cs
namespace Backend.Domain.Entity
{
    public class Message
    {
        public string Id { get; set; }
        public string Content { get; set; }
        public User Sender { get; set; }
        public User Receiver { get; set; }
        public DateTime SentTime { get; set; }
        public bool IsRead { get; set; }

        public Message(string id, string content, User sender, User receiver, DateTime sentTime, bool isRead)
        {
            Id = id;
            Content = content;
            Sender = sender;
            Receiver = receiver;
            SentTime = sentTime;
            IsRead = isRead;
        }

        public override bool Equals(object? obj)
        {
            if (obj == null)
            {
                return false;
            }
            return Equals((Message)obj);
        }

        public bool Equals(Message obj)
        {
            return obj != null && obj.Id.Equals(Id) && obj.Content.Equals(Content) && obj.Sender.Equals(Sender) && obj.Receiver.Equals(Receiver) && obj.SentTime.Equals(SentTime) && obj.IsRead.Equals(IsRead);
        }
    }
}
